crates/catalog/glue/src/catalog.rs

```diff
@@ -387,7 +387,7 @@ impl Catalog for GlueCatalog {
                 get_default_table_location(&ns, &db_name, &table_name, &self.config.warehouse)
             }
         };
-
+        creation.location = Some(location.clone());
```

Comment:
Just a minor point: this now results in two clone calls, but I don't believe either are necessary if we refactor like this:

```rust
let location = match &creation.location {
    Some(ref location) => location,
    None => {
        let ns = self.get_namespace(namespace).await?;
        let location = get_default_table_location(&ns, &db_name, &table_name, &self.config.warehouse);
        creation.location = Some(location);
        creation.location.as_ref().unwrap()
    }
};

let metadata_location = create_metadata_location(location, 0)?;
let metadata = TableMetadataBuilder::from_table_creation(creation)?
    .build()?
    .metadata;
```